From 15b9742022e33df4b66275e8c78626777b69e48e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:10 -0400
Subject: [tools] perf/record: Simplify perf_record__write

Message-id: <1408953210-26343-183-git-send-email-jolsa@redhat.com>
Patchwork-id: 88854
O-Subject: [PATCH RHEL7.1 BZ1133083 182/322] perf record: Simplify perf_record__write
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit cf8b2e6941091d2034f19c1799f8d6407bd565a4
Author: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Thu Dec 19 14:26:26 2013 -0300

description
===========
1. Since all callers either test if it is less than zero or assign its
   result to an int variable, convert it from ssize_t to int;

2. There is just one use for the 'session' variable, so use rec->session
   directly instead;

3. No need to store the result of perf_data_file__write, since that
   result is either 'size' or -1, the later making the error result to
   be stored in 'errno' and accessed thru printf's %m in the pr_err
   call.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index 8eed3d7..e8d606c 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -76,19 +76,14 @@ struct perf_record {
  long   samples;
 };
 
-static ssize_t perf_record__write(struct perf_record *rec,
-      void *buf, size_t size)
+static int perf_record__write(struct perf_record *rec, void *bf, size_t size)
 {
- struct perf_session *session = rec->session;
- ssize_t ret;
-
- ret = perf_data_file__write(session->file, buf, size);
- if (ret < 0) {
+ if (perf_data_file__write(rec->session->file, bf, size) < 0) {
   pr_err("failed to write perf data, error: %m\n");
   return -1;
  }
 
- rec->bytes_written += ret;
+ rec->bytes_written += size;
  return 0;
 }
 
-- 
1.7.1